### PR TITLE
Propz

### LINKED ISSUE #41: social media icons
try to add social media icons once again

### LINKED ISSUE #45: include year in json
use year from backend json instead if url year. 

blocked by: https://github.com/TechStreamConference/website-backend/issues/23

### LINKED ISSUE #47: check a11ly
- `user title`
- `aria-label`
- `aria-hidden`
- `alt`
- `tab-intex`

### PR DESCRIPTION
**Pull request by PurpurTentakel97**
close #47 
close #45 
close #41

### CODE REVIEW

File: src/elements/person/personRectangle.svelte [hunk 1]
```diff
@@ -58,14 +54,21 @@
 		background-color: var(--primary-color);
 		border-top-left-radius: var(--border-radius);
 		border-top-right-radius: var(--border-radius);
+		margin-bottom: 1rem;
+	}
+
+	.test {
+		flex-grow: 1;
 	}
```
Comment (peddermaster2) [inline src/elements/person/personRectangle.svelte:62]: Test successful?

File: src/elements/section/section.svelte [hunk 2]
```diff
@@ -0,0 +1,13 @@
+<script lang="ts">
+	export let id: string = '';
+	export let classes: string = '';
+</script>
+
+<section {id} class="section {classes}"><slot /></section>
+
+<style>
+	.section {
+        scroll-margin-top: 7rem;
```
Comment (peddermaster2) [inline src/elements/section/section.svelte:10]: formatting

File: src/lookup/iconLookup.ts [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    "Download": "fa-solid fa-download",
+    "Calender": "fa-solid fa-calendar-days",
+}
+
+export function iconLookup(type: string): string {
+    if (type in lookup){
+        return lookup[type];
+    }
+
+    return lookup['web'];
```
Comment (peddermaster2) [inline src/lookup/iconLookup.ts:24]:
```suggestion
    return lookup['Web'];
```

**Review by peddermaster2 — CHANGES_REQUESTED** [review]

File: src/elements/person/personRectangle.svelte (hunk 1 above)
Comment (PurpurTentakel97) [reply src/elements/person/personRectangle.svelte:62]: actually no.
see #46 

but I will rename that anyway. 🙈

File: src/elements/section/section.svelte (hunk 2 above)
Comment (PurpurTentakel97) [reply src/elements/section/section.svelte:10]: funny. In my local branch it is not like this.

File: src/lookup/iconLookup.ts (hunk 3 above)
Comment (PurpurTentakel97) [reply src/lookup/iconLookup.ts:24]: Ah thats why the fallback did not work. 🙈

**Review by peddermaster2 — APPROVED** [review]
I got no more notes 👍🏻

**Review by peddermaster2 — APPROVED** [review]

File: src/menu/impressum.ts [hunk 4]
```diff
@@ -0,0 +1,64 @@
+import type { Menu } from "types/provideTypes";
+
+export const headerOut: Menu = [
+    {
+        name: 'Start',
+        url: '/',
+        description: 'Klicke hier um zur aktuellen Hauptseite zurückzukehren'
```
Comment (mgerhold) [inline src/menu/impressum.ts:7]:
```suggestion
        description: 'Klicke hier, um zur aktuellen Hauptseite zurückzukehren'
```

File: src/menu/impressum.ts [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+    },
+    {
+        name: 'Anmelden',
+        url: '/login',
+        description: 'Klicke hier zum anmelden'
```
Comment (mgerhold) [inline src/menu/impressum.ts:12]:
```suggestion
        description: 'Klicke hier zum Anmelden'
```

File: src/menu/impressum.ts [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+    },
+    {
+        name: 'Registrieren',
+        url: '/register',
+        description: 'Klicke hier zum registrieren'
```
Comment (mgerhold) [inline src/menu/impressum.ts:17]:
```suggestion
        description: 'Klicke hier zum Registrieren'
```

File: src/menu/impressum.ts [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+    },
+];
+
+export const headerIn: Menu = [
+    {
+        name: 'Start',
+        url: '/',
+        description: 'Klicke hier um zur aktuellen Hauptseite zurückzukehren'
```
Comment (mgerhold) [inline src/menu/impressum.ts:25]:
```suggestion
        description: 'Klicke hier, um zur aktuellen Hauptseite zurückzukehren'
```

File: src/menu/impressum.ts [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+    },
+    {
+        name: 'Abmelden',
+        url: '/logout',
+        description: 'Klicke hier zum abmelden'
```
Comment (mgerhold) [inline src/menu/impressum.ts:30]:
```suggestion
        description: 'Klicke hier zum Abmelden'
```

File: src/menu/impressum.ts [hunk 9]
```diff
@@ -0,0 +1,64 @@
...
+    },
+];
+
+
+export const footerOut: Menu = [
+    {
+        name: 'Start',
+        url: '/',
+        description: 'Klicke hier um zur aktuellen Hauptseite zurückzukehren'
```
Comment (mgerhold) [inline src/menu/impressum.ts:39]:
```suggestion
        description: 'Klicke hier, um zur aktuellen Hauptseite zurückzukehren'
```

File: src/menu/impressum.ts [hunk 10]
```diff
@@ -0,0 +1,64 @@
...
+    },
+    {
+        name: 'Anmelden',
+        url: '/login',
+        description: 'Klicke hier zum anmelden'
```
Comment (mgerhold) [inline src/menu/impressum.ts:44]:
```suggestion
        description: 'Klicke hier zum Anmelden'
```

File: src/menu/impressum.ts [hunk 11]
```diff
@@ -0,0 +1,64 @@
...
+    },
+    {
+        name: 'Registrieren',
+        url: '/register',
+        description: 'Klicke hier zum registrieren'
```
Comment (mgerhold) [inline src/menu/impressum.ts:49]:
```suggestion
        description: 'Klicke hier zum Registrieren'
```

File: src/menu/impressum.ts [hunk 12]
```diff
@@ -0,0 +1,64 @@
...
+    },
+];
+
+export const footerIn: Menu = [
+    {
+        name: 'Start',
+        url: '/',
+        description: 'Klicke hier um zur aktuellen Hauptseite zurückzukehren'
```
Comment (mgerhold) [inline src/menu/impressum.ts:57]:
```suggestion
        description: 'Klicke hier, um zur aktuellen Hauptseite zurückzukehren'
```

File: src/menu/impressum.ts [hunk 13]
```diff
@@ -0,0 +1,64 @@
...
+    },
+    {
+        name: 'Intern',
+        url: '/intern',
+        description: 'Klicke hier um in den internen Bereich zu gelangen'
```
Comment (mgerhold) [inline src/menu/impressum.ts:62]:
```suggestion
        description: 'Klicke hier, um in den internen Bereich zu gelangen'
```

File: src/menu/login.ts
Comment (mgerhold) [inline src/menu/login.ts:1]: I don't want to spam too many comments, but this and also other files contain the same or similar typos like the previous file.

File: src/routes/impressum/+page.svelte
Comment (mgerhold) [inline src/routes/impressum/+page.svelte:1]: I just realise that "impressum" is German 😅 It should be "imprint" instead (just for consistency).

File: src/routes/impressum/+page.svelte [hunk 14]
```diff
@@ -1,85 +1,76 @@
...
-		<SpacerOneLine />
+		<TextLine classes="impressum-one-line-margin">14727 Premnitz OT Mögelin</TextLine>
 		<TextLine>
 			E-Mail:
-			<Link classes="inline" href="mailto:info@test-conf.de" newTab={true}>info@test-conf.de</Link>
+			<Link
+				classes="inline"
+				href="mailto:info@test-conf.de"
+				newTab={true}
+				title="Klicke hier um uns eine Mail zu schreiben">info@test-conf.de</Link
```
Comment (mgerhold) [inline src/routes/impressum/+page.svelte:32]:
```suggestion
				title="Klicke hier, um uns eine Mail zu schreiben">info@test-conf.de</Link
```

File: src/routes/impressum/+page.svelte [hunk 15]
```diff
@@ -1,85 +1,76 @@
...
 			Online-Streitbeilegungs-Plattform ist hier zu finden:
-			<Link classes="inline" href="http://ec.europa.eu/odr" newTab={true}>
-				http://ec.europa.eu/odr</Link
-			>. Unsere E-Mail lautet:
-			<Link classes="inline" href="mailto:info@test-conf.de" newTab={true}>info@test-conf.de</Link>
+			<Link
+				classes="inline"
+				href="http://ec.europa.eu/odr"
+				newTab={true}
+				title="Klicke hier um dich über die Europäische
```
Comment (mgerhold) [inline src/routes/impressum/+page.svelte:50]:
```suggestion
				title="Klicke hier, um dich über die Europäische
```

File: src/routes/impressum/+page.svelte [hunk 16]
```diff
@@ -1,85 +1,76 @@
...
+			Online-Streitbeilegungs-Plattform zu informieren"
+			>
+				http://ec.europa.eu/odr
+			</Link>
+			. Unsere E-Mail lautet:
+			<Link
+				classes="inline impressum-one-line-margin"
+				href="mailto:info@test-conf.de"
+				newTab={true}
+				title="Klicke hier um uns eine Mail zu schreiben"
```
Comment (mgerhold) [inline src/routes/impressum/+page.svelte:60]:
```suggestion
				title="Klicke hier, um uns eine Mail zu schreiben"
```

File: src/routes/impressum/+page.svelte [hunk 17]
```diff
@@ -1,85 +1,76 @@
...
 		<Paragraph>
 			Wir weisen aber darauf hin, dass wir nicht bereit sind, uns am Streitbeilegungsverfahren im
 			Rahmen der Europäischen Online-Streitbeilegungs-Plattform zu beteiligen. Nutzen Sie zur
 			Kontaktaufnahme bitte unsere obige E-Mail und Telefonnummer.
 		</Paragraph>
 
-		<Spacer --height="5rem" />
-		<SubHeadline>Disclaimer – rechtliche Hinweise</SubHeadline>
-		<Spacer --height="2rem" />
+		<SubHeadline classes="imperssum-subheadline">Disclaimer – rechtliche Hinweise</SubHeadline>
```
Comment (mgerhold) [inline src/routes/impressum/+page.svelte:71]:
```suggestion
		<SubHeadline classes="impressum-subheadline">Disclaimer – rechtliche Hinweise</SubHeadline>
```

or rather

`imprint-subheadline`

File: src/routes/impressum/+page.svelte [hunk 18]
```diff
@@ -114,23 +103,29 @@
...
-		<TextLine
-			>Die Darstellung dieser Website in fremden Frames ist nur mit schriftlicher Erlaubnis
-			zulässig.
+		<TextLine classes="impressum-one-line-margin">
+			Die Darstellung dieser Website in fremden Frames ist nur mit schriftlicher Erlaubnis zulässig.
 		</TextLine>
-		<SpacerOneLine />
-		<Paragraph>
+		<Paragraph classes="impressum-last-paragraph">
 			§ 4 Besondere Nutzungsbedingungen Soweit besondere Bedingungen für einzelne Nutzungen dieser
```
Comment (mgerhold) [inline src/routes/impressum/+page.svelte:110]: Shouldn't there be a linebreak after "Nutzungsbedingungen"?

File: src/routes/impressum/+page.svelte [hunk 19]
```diff
@@ -114,23 +103,29 @@
...
 			Website von den vorgenannten Paragraphen abweichen, wird an entsprechender Stelle ausdrücklich
 			darauf hingewiesen. In diesem Falle gelten im jeweiligen Einzelfall die besonderen
 			Nutzungsbedingungen. Quelle:
-			<Link classes="inline" href="https://www.juraforum.de/" newTab={true}>Hier ansehen >></Link>
+			<Link
+				classes="inline"
+				href="https://www.juraforum.de/"
+				newTab={true}
+				title="Klicke hier um dich über besondere Nutzungsbedingungen zu informieren"
```
Comment (mgerhold) [inline src/routes/impressum/+page.svelte:118]:
```suggestion
				title="Klicke hier, um zur Quelle dieses Online-Impressums zu gelangen"
```

File: src/routes/register/+page.svelte [hunk 20]
```diff
@@ -143,69 +132,77 @@
...
-				<Spacer --height="2rem" />
+				</div>
 				<Input
+					classes="register-basic-input-line"
 					id="register-username"
 					type="text"
 					labelText="Name:"
 					placeholderText="Name"
 					bind:value={username}
+					ariaLabel="Gebe den Nutzernamen ein"
```
Comment (mgerhold) [inline src/routes/register/+page.svelte:158]:
```suggestion
					ariaLabel="Gib den Nutzernamen ein"
```

File: src/routes/register/+page.svelte [hunk 21]
```diff
@@ -143,69 +132,77 @@
...
 				/>
-				<Spacer --height="1rem" />
 				<Input
+					classes="register-basic-input-line"
 					id="register-email"
 					type="email"
 					labelText="E-Mail:"
 					placeholderText="E-Mail"
 					bind:value={email}
+					ariaLabel="Gebe die E-Mail ein"
```
Comment (mgerhold) [inline src/routes/register/+page.svelte:170]:
```suggestion
					ariaLabel="Gib die E-Mail ein"
```

File: src/routes/register/+page.svelte [hunk 22]
```diff
@@ -143,69 +132,77 @@
...
 				/>
-				<Spacer --height="1rem" />
 				<Input
+					classes="register-basic-input-line"
 					id="register-password"
 					type="password"
 					labelText="Passwort:"
 					placeholderText="Passwort"
 					bind:value={password_1}
+					ariaLabel="Gebe das passwort ein"
```
Comment (mgerhold) [inline src/routes/register/+page.svelte:182]:
```suggestion
					ariaLabel="Gib das Passwort ein"
```

File: src/routes/register/+page.svelte [hunk 23]
```diff
@@ -143,69 +132,77 @@
...
 				/>
-				<Spacer --height="1rem" />
 				<Input
+					classes="register-basic-input-line"
 					id="register-password-repeat"
 					type="password"
 					labelText="Passwort wiederholen:"
 					placeholderText="Passwort wiederholen"
 					bind:value={password_2}
+					ariaLabel="Wiederhole das Password"
```
Comment (mgerhold) [inline src/routes/register/+page.svelte:192]:
```suggestion
					ariaLabel="Wiederhole das Passwort"
```

File: src/routes/register/+page.svelte [hunk 24]
```diff
@@ -143,69 +132,77 @@
...
 					on:input={startPasswordTimer}
 				/>
-				<Spacer --height="3rem" />
-				<Button type={'submit'}>Registrieren</Button>
+				<Button
```
Comment (mgerhold) [inline src/routes/register/+page.svelte:195]: I think there's too much free space between the input fields and the button.

File: src/routes/register/+page.svelte [hunk 25]
```diff
@@ -215,32 +212,29 @@
...
+				</TextLine>
 				<Paragraph --text-align="center">
 					Du bekommst jetzt eine Mail. Bestätige deine Mailadresse, um dich anmelden zu können.
 					Weitere Infos findest du nach dem Anmelden.
 				</Paragraph>
-				<Spacer --height="6rem" />
 				<div class="button-wrapper">
 					<a href="/">
-						<Button>Start</Button>
+						<Button ariaLabel="Klicke um zur aktuellen Hauptseite zu gelangen">Start</Button>
```
Comment (mgerhold) [inline src/routes/register/+page.svelte:227]:
```suggestion
						<Button ariaLabel="Klicke, um zur aktuellen Hauptseite zu gelangen">Start</Button>
```

File: src/routes/register/+page.svelte [hunk 26]
```diff
@@ -215,32 +212,29 @@
...
 					</a>
 					<a href="./login">
-						<Button>Anmelden</Button>
+						<Button ariaLabel="Klicke um dich anzumelden">Anmelden</Button>
```
Comment (mgerhold) [inline src/routes/register/+page.svelte:230]:
```suggestion
						<Button ariaLabel="Klicke, um dich anzumelden">Anmelden</Button>
```

**Review by mgerhold — CHANGES_REQUESTED** [review]
Grammar check completed 😅

File: src/menu/login.ts
Comment (PurpurTentakel97) [reply src/menu/login.ts:1]: I hope I found all. 🙈

File: src/menu/menuItems.ts [hunk 27]
```diff
@@ -0,0 +1,38 @@
...
+export const intern: MenuItem = {
+    name: 'Intern',
+    url: '/backend',
+    description: "Klicke, um zum internen Bereich zu navigieren"
+}
+
+export const logout: MenuItem = {
+    name: 'Abmelden',
+    url: '/logout',
+    description: 'Klicke, um zum Abmelden'
```
Comment (mgerhold) [inline src/menu/menuItems.ts:19]:
```suggestion
    description: 'Klicke zum Abmelden'
```

File: src/menu/menuItems.ts [hunk 28]
```diff
@@ -0,0 +1,38 @@
...
+}
+
+export const login: MenuItem = {
+    name: 'Anmelden',
+    url: '/login',
+    description: 'Klicke, um zum Anmelden'
```
Comment (mgerhold) [inline src/menu/menuItems.ts:25]:
```suggestion
    description: 'Klicke zum Anmelden'
```

File: src/menu/menuItems.ts [hunk 29]
```diff
@@ -0,0 +1,38 @@
...
+}
+
+export const register: MenuItem = {
+    name: 'Registrieren',
+    url: '/register',
+    description: 'Klicke, zum registrieren'
```
Comment (mgerhold) [inline src/menu/menuItems.ts:31]:
```suggestion
    description: 'Klicke zum Registrieren'
```

File: src/menu/menuItems.ts [hunk 30]
```diff
@@ -0,0 +1,38 @@
...
+}
+
+export const imprint: MenuItem = {
+    name: 'Impressum',
+    url: '/imprint',
+    description: 'Klicke, um zum impressum zu gelangen'
```
Comment (mgerhold) [inline src/menu/menuItems.ts:37]:
```suggestion
    description: 'Klicke, um zum Impressum zu gelangen'
```

File: src/routes/login/+page.svelte [hunk 31]
```diff
@@ -77,13 +77,13 @@
...
-				ariaLabel="Gebe das Passwort ein"
+				ariaLabel="Gib das Passwort ein"
 				bind:value={password}
 			/>
-			<Link href="/register" title="Klicke um einen neuen Account anzulegen">
+			<Link href="/register" title="Klicke, um einen neuen Account anzulegen">
 				Noch keinen Account?
 			</Link>
-			<Button classes="text login-submit-button" type="submit" ariaLabel="Klicke zum anmelden">
+			<Button classes="text login-submit-button" type="submit" ariaLabel="Klicke, zum Anmelden">
```
Comment (mgerhold) [inline src/routes/login/+page.svelte:86]:
```suggestion
			<Button classes="text login-submit-button" type="submit" ariaLabel="Klicke zum Anmelden">
```

File: src/routes/register/+page.svelte [hunk 32]
```diff
@@ -189,13 +189,13 @@
...
 					bind:value={password_2}
-					ariaLabel="Wiederhole das Password"
+					ariaLabel="Wiederhole das Passwort"
 					on:input={startPasswordTimer}
 				/>
 				<Button
 					classes="text register-submit-button"
 					type={'submit'}
-					ariaLabel="Klicke zum Registrieren"
+					ariaLabel="Klicke, zum Registrieren"
```
Comment (mgerhold) [inline src/routes/register/+page.svelte:198]:
```suggestion
					ariaLabel="Klicke zum Registrieren"
```

**Review by mgerhold — CHANGES_REQUESTED** [review]

**Review by mgerhold — APPROVED** [review]

**Pull request merged**